code/game/objects/items/carp_lasso.dm

```diff
@@ -7,53 +7,70 @@
 	var/timer
 	///Ref to lasso'd carp
 	var/mob/living/simple_animal/mob_target
+	///Range we can lasso things at
+	var/range = 8
 	///Whitelist of allowed animals
-	var/list/whitelist_mobs = list(/mob/living/simple_animal/hostile/carp, /mob/living/simple_animal/cow, /mob/living/simple_animal/hostile/retaliate/dolphin)
+	var/list/whitelist_mobs
+
+/obj/item/mob_lasso/Initialize(mapload)
+	. = ..()
+	whitelist_mobs = typecacheof(list(/mob/living/simple_animal/hostile/carp, /mob/living/simple_animal/cow, /mob/living/simple_animal/hostile/retaliate/dolphin), only_root_path = TRUE)
 
 /obj/item/mob_lasso/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	. = ..()
-	if(isliving(target) && check_allowed(target) && !iscarbon(target) && !issilicon(target) && locate(target) in oview(9, get_turf(src)))
-		var/mob/living/simple_animal/C = target
-		if(IS_DEAD_OR_INCAP(C))
-			to_chat(user, "<span class='warning'>[target] is dead.</span>")
+	var/failed = FALSE
+	if(!isliving(target))
+		failed = TRUE
+	if(!check_allowed(target))
+		failed = TRUE
+	if(iscarbon(target) || issilicon(target))
+		failed = TRUE
+	if(!(locate(target) in oview(range, user)))
+		if(ismob(target))
+			to_chat(user, "<span class='warning'>you can't lasso [target] from here!</span>")
+		return
+	if(failed)
+		if(ismob(target))
```

Comment:
why do these check ismob(target)? can you even lasso something that is not a mob? you already fail if the target is not living

Reply:
This is done so that you don't end up with "The floor seems a bit big for this..."x8 when you're trying to lasso something. It's something you can see happen on live if you play around with lassos.